uber/models.py

```diff
@@ -856,6 +862,7 @@ def model_mixin(cls, model):
 
 
 class Group(MagModel, TakesPaymentMixin):
+    public_id    = Column(UUID, default=lambda: str(uuid4()))
```

Comment:
Because I personally view "blindly trust uuids" as one of the system's greatest strengths rather than a weakness that needs to be fixed.  After slightly over 100,000 registrations under this system I'm more than willing to defend this practice :)

The long run plan is to add an optional flag on people's accounts that allows them to turn on authentication if they want, which would be off by default for MAGFest and probably on by default for conventions which don't believe you can create 100,000 registrations safely in this manner.